framework/src/modules/chain/chain.js

```diff
@@ -272,6 +295,40 @@ module.exports = class Chain {
 					action.params[0],
 					action.params[1]
 				),
+			blocks: async action =>
```

Comment:
I would suggest to pass the requested number of blocks as a variable, or returning a dynamic number depending on the sizes of the requested blocks.

Reply:
Yes I think as a new issue.

Reply:
@jondubois @diego-G same as above, we don't want to change the behavior now, so please create a new issue, and we will tackle later